### DISCUSSION TITLE
failed to publish artifacts to private Gitlab with ENV GITLAB_TOKEN

### DISCUSSION THREAD

**Discussion by zhonghuaxunGM** [goreleaser/goreleaser#2505] [Q&A] [answered]
I USE PRIVATE GITLAB. 
I chose one of the Personal Access Tokens in  User Settings .  It's Scope  included api, read_user, read_repository, write_repository .

I ALSO HAS Set my Environment Variable.  

```
$ export GITLAB_TOKEN=Ws2syxxxxxxxxxxxPxmsy
$ env|grep GIT
GITLAB_TOKEN=Ws2syxxxxxxxxxxxxmsy
```

ALSO I SET Personal Access Tokens  in configuration file which is located on /c/Users/zhxu/.config/goreleaser/gitlab_token 

LOG:

```
         • creating or updating release repo= tag=v1.0.1
   ⨯ release failed after 7.75s error=github/gitlab/gitea releases: failed to publish artifacts: GET https://gitlab.com/api/v4/projects/xxxx/xxxxxx/releases/v1.0.1: 401 {message: 401 Unauthorized}
```

**Comment by zhonghuaxunGM** [comment]
i  suppose  the message  GET https://gitlab.com/api/v4/projects/xxxx/xxxxxx . but actually my registery is http://gitlab.xxxxxx.org/xxx/xxx

**Comment by caarlos0** [comment]
did you set `gitlab_urls` according to [docs](https://goreleaser.com/scm/gitlab/)? 

judging by the message, that's what is missing.

---

EDIT: similar issue https://github.com/goreleaser/goreleaser/issues/2356

**Comment by zhonghuaxunGM** [comment, marked as answer]
If I set the following content in the `.goreleaser.yaml`, it works ! 

```yaml
gitlab_urls:
  api: http://<host>/api/v4
  download: http://<host>
  skip_tls_verify: true
```
